File: 2022/day04/day04.jl
function main()
    data = filter(x -> !isempty(x), split(readchomp("input.txt"), "\n"))

    mapreduce(+, data) do x
        m = parse.(Int, split(x, ('-', ',')))
        a, b = m[1]:m[2], m[3]:m[4]

        (a ⊆ b) | (b ⊆ a)  # part 1
        # !isdisjoint(a, b)  # part 2
    end
end

main()

File: 2022/day04/day04.py
import re

data = open("input.txt").read().strip().split()

n = 0
for line in data:
    m = list(map(int, re.findall(r"(\d+)-(\d+),(\d+)-(\d+)", line)[0]))
    a = range(m[0], m[1] + 1)
    b = range(m[2], m[3] + 1)

    # part 1
    if all([ai in b for ai in a]) or all([bi in a for bi in b]):
        n += 1

    # # part 2
    # if any([ai in b for ai in a]) or any([bi in a for bi in b]):
    #     n += 1
print(n)
